core-chart/templates/argocd/setup.yaml

```diff
@@ -4,105 +4,46 @@ kind: ConfigMap
 metadata:
   name: "{{ .Release.Name }}-argocd-kubestellar-setup"
 data:
-  # This script is used by the "{{ .Release.Name }}-argocd-kubestellar-setup" job to
-  # 1. register KubeStellar WDSes as Argo CD target clusters
-  # 2. create any Argo CD application listed in KubeStelar values
+  # This script is used by the "{{ .Release.Name }}-argocd-kubestellar-setup" job to:
+  # 1. Register KubeStellar WDSes as Argo CD target clusters
+  # 2. Create any Argo CD applications listed in KubeStellar values
+  # Note: Waiting logic for dependencies moved to initContainers for better reliability

```

Comment:
FYI, IMO, this sort of comment belongs in the PR, not the file that everybody sees afterward.